### PR TITLE
new SUPG parameter design

### PR DESCRIPTION
**Pull request by tjhei**
This is the more robust parameter design as discussed with @tcclevenger over the last few days.

Synthetic benchmarks show little to no difference, but I expect it to be more robust if one of the parameters (\|u\|, eps) goes to zero.

### CODE REVIEW

**Comment by tjhei** [comment]
@tcclevenger can you check if this makes sense to you?

**Comment by tjhei** [comment]
Note, #2999 produces somewhat different results for the max temperature:
image

File: source/simulator/entropy_viscosity.cc [hunk 1]
```diff
@@ -599,14 +596,32 @@ namespace aspect
...
+              {
+                // Diffusion dominant case, no stabilization needed:
+                viscosity_per_cell[cell->active_cell_index()] = 0.0;
+              }
+            else
+              {
+                // To avoid a division by zero, increase eps slightly. The actual value is not
+                // important, as long as the result is still a valid number. Note that this
+                // is only important if \|u\| and eps are zero.
+                const double peclet = peclet_times_eps / (eps + 1e-100);
```
Comment (tcclevenger) [inline source/simulator/entropy_viscosity.cc]: Do we need `eps+1e-100` here? If `eps=0`, the `peclet_times_eps==0.0` above will be true, correct? I guess it doesn't really matter though...

File: source/simulator/entropy_viscosity.cc [hunk 2]
```diff
@@ -599,14 +596,32 @@ namespace aspect
...
+                const double coth_of_peclet = (1.0 + exp(-2.0*peclet)) / (1.0 - exp(-2.0*peclet));
+                const double delta = h/2.0/norm_of_advection_term * (coth_of_peclet - 1.0/peclet);
```
Comment (tcclevenger) [inline source/simulator/entropy_viscosity.cc]:
```suggestion
                const double delta = h/2.0/norm_of_advection_term/fe_order * (coth_of_peclet - 1.0/peclet);
```

Comment (tjhei) [reply source/simulator/entropy_viscosity.cc]: Wow, good catch. Thanks!

**Comment by tjhei** [comment]
Okay, now this PR passes the tests without a change in the results! :+1:

**Comment by tcclevenger** [comment]
Really great work Timo!

**Comment by tjhei** [comment]
with this fixed version and #3020 I am now getting the same results (better than the old SUPG):
image

**Comment by tjhei** [comment]
With this + #3020:
image

**Comment by gassmoeller** [comment]
Do you want me to merge the current version? Looks reasonable to me.

**Comment by tjhei** [comment]
> Looks reasonable to me.

yes.

**Pull request merged**